datafusion/core/src/datasource/listing/table.rs

```diff
@@ -982,18 +980,6 @@ impl TableProvider for ListingTable {
                     return Ok(TableProviderFilterPushDown::Exact);
                 }
 
-                // if we can't push it down completely with only the filename-based/path-based
-                // column names, then we should check if we can do parquet predicate pushdown
-                let supports_pushdown = self.options.format.supports_filters_pushdown(
-                    &self.file_schema,
-                    &self.table_schema,
-                    &[filter],
-                )?;
-
-                if supports_pushdown == FilePushdownSupport::Supported {
-                    return Ok(TableProviderFilterPushDown::Exact);
-                }
```

Comment:
The point of this PR is that this moves from being something specialized that `ListingTable` does to anything that works for any TableProvider / they don't need to do anything special! The checks for compatibility also happen all within the parquet data source machinery, instead of leaking implementations via `supports_filters_pushdown`.

Reply:
I think many users of DataFusion (based on our usage, talks I've seen and examples we have) use custom `TableProvider` implementations.

I would keep `supports_filters_pushdown` so that `TableProviders` can do `Exact` pruning of filters, e.g. using partition columns.

Reply:
> We can justify implementing other TableProviders for Parquet, but still I cannot understand why we need to degrade the capabilities of our ListingTable. Is't it always better pruning/simplifying things at the higher levels as possible?

I don't think this degrades the capabilities of the current listing table. I think the only implications are for anyone who used a custom `FileFormat` and impleented `supports_filters_pushdown` -- I suspect this is not very common and we can likely avoid consternation by mentioning it in the upgrade guide (see comment below)